grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details.ts

```diff
@@ -31,36 +27,53 @@ export class OsqueryDetails extends Plugin {
     map(singleItem => singleItem.payload as OsqueryResult),
   );
 
-  readonly resultsTable$ = this.osqueryResults$.pipe(
-    map(result => result?.table),
-  );
-
-  readonly resultsStderr$ = this.osqueryResults$.pipe(
-    map(result => result?.stderr),
-  );
-
   readonly osqueryProgress$: Observable<OsqueryProgress> = this.flowListEntry$.pipe(
     map(listEntry => listEntry.flow.progress as OsqueryProgress),
     filter(isNonNull),
   );
 
+  readonly resultsTable$ = this.osqueryResults$.pipe(
+    map(result => result.table),
+    filter(isNonNull),
+  );
+
   readonly progressTable$ = this.osqueryProgress$.pipe(
     map(progress => progress.partialTable),
+    filter(isNonNull),
   );
 
-  readonly additionalRowsAvailable$ = this.osqueryProgress$.pipe(
-    map(progress => {
-      const progressTableLength = progress.partialTable?.rows?.length;
-      const fullTableLength = progress.totalRowCount;
+  readonly displayTable$ = concat(
+    this.progressTable$.pipe(takeUntil(this.resultsTable$)),
+    this.resultsTable$,
+  );
+
+  readonly totalAndDisplayedRowCount = combineLatest([
```

Comment:
`combineLatest` only emits if all inner observables also emitted at least once. If this is unexpected, use `startWith()` to provide a starting value.